### PR TITLE
Complete merchant invoice user stories

### LINKED ISSUE #23: Merchant Invoice Show Page: Update Item Status
As a merchant
When I visit my merchant invoice show page
I see that each invoice item status is a select field
And I see that the invoice item's current status is selected
When I click this select field,
Then I can select a new status for the Item,
And next to the select field I see a button to "Update Item Status"
When I click this button
I am taken back to the merchant invoice show page
And I see that my Item's status has now been updated

### LINKED ISSUE #24: Merchant Invoice Show Page: Total Revenue
As a merchant
When I visit my merchant invoice show page
Then I see the total revenue that will be generated from all of my items on the invoice

### PR DESCRIPTION
**Pull request by mkrumholz**
## Changes

- Removed some comments
- Fixed some spacing
- Added tests for nav partials 
- Added nav partials to each page
- Created functionality to update invoice item status
- Refactored a bunch of active record

## Special review requests

- Just make sure I didn't update/remove anything you wanted in there!

## Checklist
- [x] Tests written where required
- [x] All tests passing
- [x] SimpleCov at 100%

closes #23 
closes #24

### CODE REVIEW

**Review by aliyamerali — APPROVED** [review]
All changes look good on my end!

File: app/views/admin/index.html.erb [hunk 1]
```diff
@@ -1,11 +1,6 @@
-<h1 id="dash">Admin Dashboard</h1>
-
-
-<ul>
-  <li><%= link_to "Merchants Index", '/admin/merchants' %></li>
-  <li><%= link_to "Invoices Index", '/admin/invoices' %></li>
-</ul>
+<%= render partial: './partials/admin_nav' %>
```
Comment (Sidarth20) [inline app/views/admin/index.html.erb:1]: Thanks for adding the partials to all the pages!

File: spec/features/admin/invoices/show_spec.rb [hunk 2]
```diff
@@ -7,7 +7,7 @@
     @merchant_1 = Merchant.create!(name: "Ralph's Monkey Hut")
     @item_1 = @merchant_1.items.create!(name: 'Pogs', description: 'Stack of pogs.', unit_price: 500,)
     @item_2 = @merchant_1.items.create!(name: 'Frog statue', description: 'Statue of a frog', unit_price: 10000,)
-    @item_3 = @merchant_1.items.create!(name: 'Rabid Woliverine', description: 'No refunds', unit_price: 10,)
+    @item_3 = @merchant_1.items.create!(name: 'Rabid Wolverine', description: 'No refunds', unit_price: 10,)
```
Comment (Sidarth20) [inline spec/features/admin/invoices/show_spec.rb:10]: Nice spelling catch!

File: spec/features/merchants/invoices/show_spec.rb [hunk 3]
```diff
@@ -36,7 +36,7 @@
       visit "/merchants/#{merchant.id}/invoices/#{invoice_1.id}"
 
       expect(page).to have_content(invoice_1.id)
-      expect(page).to have_content(invoice_1.status)
+      expect(page).to have_select('invoice_item[status]', selected: "Packaged")
```
Comment (Sidarth20) [inline spec/features/merchants/invoices/show_spec.rb:39]: Nice use of have_select to test the user story option

File: spec/features/merchants/invoices/show_spec.rb [hunk 4]
```diff
@@ -83,8 +83,73 @@
...
+
+      invoice_1 = customer_1.invoices.create!(status: 1, created_at: "2012-03-06 14:54:15 UTC")
+      invoice_2 = customer_2.invoices.create!(status: 1, created_at: "2012-03-09 14:54:15 UTC")
+      
+      # items for invoice 1
+      invoice_item_1 = InvoiceItem.create!(quantity: 2, unit_price: 10000, item_id: item_1.id, invoice_id: invoice_1.id, status: 1) # $200
+      invoice_item_2 = InvoiceItem.create!(quantity: 2, unit_price: 5000, item_id: item_2.id, invoice_id: invoice_1.id, status: 1) # $100
+      invoice_item_3 = InvoiceItem.create!(quantity: 2, unit_price: 1000,item_id: item_3.id, invoice_id: invoice_1.id, status: 1) # $20
+      invoice_item_5 = InvoiceItem.create!(quantity: 2, unit_price: 3000,item_id: item_5.id, invoice_id: invoice_1.id, status: 1) # Other merchant rev
+      invoice_item_6 = InvoiceItem.create!(quantity: 2, unit_price: 2000,item_id: item_6.id, invoice_id: invoice_1.id, status: 2) # Other merchant rev
```
Comment (Sidarth20) [inline spec/features/merchants/invoices/show_spec.rb:115]: Great job with this test & making sure that the revenue doesn't pull from the 2nd merchant!

File: spec/features/merchants/invoices/show_spec.rb [hunk 5]
```diff
@@ -83,8 +83,73 @@
...
+        expect(page).to have_select('invoice_item[status]', selected: "Packaged")
+
+        select 'Shipped', from: 'invoice_item[status]'
+        click_button 'Update'
+      end
+
+      expect(current_path).to eq "/merchants/#{merchant.id}/invoices/#{invoice_1.id}"
+
+      within "tr#ii-#{invoice_item_1.id}" do
+        expect(page).to have_select('invoice_item[status]', selected: "Shipped")
```
Comment (Sidarth20) [inline spec/features/merchants/invoices/show_spec.rb:151]: Great job using within testing for updating the item's status!!!

File: spec/features/partials/admin_nav_spec.rb [hunk 6]
```diff
@@ -0,0 +1,30 @@
+require 'rails_helper'
```
Comment (Sidarth20) [inline spec/features/partials/admin_nav_spec.rb:1]: Thanks for the writing the test for the partials!

File: app/models/invoice.rb [hunk 7]
```diff
@@ -17,22 +17,21 @@ def self.unshipped_items
...
-    .group(:updated_at)
-    .limit(1)
-    .pluck(:updated_at)
-  end
-
   def item_sale_price
-    self.items.joins(:invoice_items).select('items.*, invoice_items.unit_price as sale_price, invoice_items.quantity as sale_quantity')
+    items
+    .joins(:invoice_items)
+    .select('items.*, invoice_items.unit_price as sale_price, invoice_items.quantity as sale_quantity')
```
Comment (Sidarth20) [inline app/models/invoice.rb:23]: @mkrumholz This is probably a question for the view, but did you needed to create all of these queries for the total revenue?

**Review by Sidarth20 — APPROVED** [review]

**Pull request merged**